tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/core/NN.java

```diff
@@ -0,0 +1,379 @@
+import org.tensorflow.op.math.*;
+import org.tensorflow.op.nn.raw.SoftmaxCrossEntropyWithLogits;
+import org.tensorflow.op.nn.raw.SparseSoftmaxCrossEntropyWithLogits;
+import org.tensorflow.types.*;
+import org.tensorflow.types.family.TNumber;
+import org.tensorflow.op.dtypes.Cast;
+import org.tensorflow.types.family.TType;
+import org.tensorflow.op.linalg.Transpose;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+@Operator(group = "nn")
+public abstract class NN {
+
+  /**
+   * Computes softmax cross entropy between `logits` and `labels`.
+   *
+   * <p>Measures the probability error in discrete classification tasks in which the classes are
+   * mutually exclusive (each entry is in exactly one class). For example, each CIFAR-10 image is
+   * labeled with one and only one label: an image can be a dog or a truck, but not both.
+   *
+   * <p>**NOTE:** While the classes are mutually exclusive, their probabilities need not be. All
+   * that is required is that each row of `labels` is a valid probability distribution. If they are
+   * not, the computation of the gradient will be incorrect.
+   *
+   * <p>If using exclusive `labels` (wherein one and only one class is true at a time), see
+   * `sparse_softmax_cross_entropy_with_logits`.
+   *
+   * <p>Usage:
+   *
+   * <pre>
+   *   >>> logits = [[4.0, 2.0, 1.0], [0.0, 5.0, 1.0]]
+   *   >>> labels = [[1.0, 0.0, 0.0], [0.0, 0.8, 0.2]]
+   *   >>> tf.nn.softmax_cross_entropy_with_logits(labels=labels, logits=logits)
+   *   <tf.Tensor: shape=(2,), dtype=float32,
+   *   numpy=array([0.16984604, 0.82474494], dtype=float32)>
+   * </pre>
+   *
+   * <p>Backpropagation will happen into both `logits` and `labels`. To disallow backpropagation
+   * into `labels`, pass label tensors through `tf.stop_gradient` before feeding it to this
+   * function.
+   *
+   * @param scope current scope
+   * @param labels Each vector along the class dimension should hold a valid probability
+   *     distribution e.g. for the case in which labels are of shape `[batch_size, num_classes]`,
+   *     each row of `labels[i]` must be a valid probability distribution.
+   * @param logits Per-label activations, typically a linear output. These activation energies are
+   *     interpreted as unnormalized log probabilities.
+   * @param axis The class dimension. -1 is the last dimension.
+   * @param <U> the data type of the logits
+   * @param <T> the number type of the operands
+   * @return the softmax cross entropy loss. Its type is the same as `logits` and its shape is the
+   *     same as `labels` except that it does not have the last dimension of `labels`.
+   */
+  @Endpoint(name = "softmaxCrossEntropyWithLogits")
+  public static <U extends TType, T extends TNumber> Operand<T> softmaxCrossEntropyWithLogits(
+      Scope scope, Operand<T> labels, Operand<U> logits, int axis) {
+    axis = axis % logits.asOutput().shape().numDimensions();
```

Comment:
Maybe you should create a subscope like you do in `sparseSoftmaxCrossEntropyWithLogits`?

Reply:
OK, also did it in SigmoidCrossEntropyWithLogits